samza-api/src/main/java/org/apache/samza/operators/windows/internal/WindowInternal.java

```diff
@@ -55,17 +56,23 @@
    */
   private final Function<M, Long> eventTimeExtractor;
 
-  private Trigger earlyTrigger;
+  private final WindowType windowType;
```

Comment:
For ones that probably need explanation (for instance, `initialValueSupplier` and `windowType`, I ended up adding javadocs. I refrained from adding javadocs for obvious ones like `earlyTrigger`, `lateTrigger` and `defaultTrigger`, since they are rather superfluous.